snuba/query/mql/parser.py

```diff
@@ -304,6 +335,17 @@ def parse_mql_query_initial(
     processors and are supposed to update the AST.
     """
     try:
+        """
+        Example of parsed tree for:
```

Comment:
That was my approach initially. The reason I decided to return a dictionary instead is because of the indexer resolved names. It's easier to find and replace strings in that dictionary, then create the AST with those arguments , rather than instantiating the AST object and traversing + replacing fields. What do you think?

Reply:
Updated `visit_` functions to return `Expressions` where possible. However, I think it's better for the top-level visitor to return dictionary as the information (like metric_name) still needs to be used outside of the visitor class.